zerver/views/registration.py

```diff
@@ -436,6 +438,11 @@ def accounts_register(
                 acting_user=None,
             )
 
+        if enable_marketing_emails:
+            do_marketing_subscribe(user_profile)
+        else:
+            do_marketing_unsubscribe(user_profile)
```

Comment:
We should avoid extra database queries by plumbing the parameter through the `do_create_user` rather than doing a change action just after user creation.

(We don't do this with `bulk_add_subscription`, below, because it creates objects in another table, rather than just setting a field in an existing one).